### PR TITLE
io file reviews

### PR DESCRIPTION
**Pull request by nabobalis**
IO TIME.

### CODE REVIEW

**Comment by nabobalis** [comment]
`403 Forbidden` for the docbuild, probably need to update a url.

File: sunpy/io/ana.py [hunk 1]
```diff
@@ -1,29 +1,20 @@
 """
-ANA File Reader
+This module provides an ANA file Reader.
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy/io/ana.py]: This is a topic that we need to discuss, what if anything do we put at the top of python files.

File: sunpy/io/ana.py [hunk 2]
```diff
@@ -1,29 +1,20 @@
...
-
-Notes
------
-ANA is a script that allows people to access compressed ana files.
-It accesses a C library, based on Michiel van Noort's
-IDL DLM library 'f0' which contains a cleaned up version of the original
-anarw routines.
 
-Created by Tim van Werkhoven (t.i.m.vanwerkhoven@gmail.com) on 2009-02-11.
-Copyright (c) 2009--2011 Tim van Werkhoven.
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy/io/ana.py]: Sure. I'll add it back.

File: sunpy/io/ana.py [hunk 3]
```diff
@@ -82,29 +72,30 @@ def get_header(filename, debug=False):
...
     data : `numpy.ndarray`
-        Name of data to be stored.
+        The data to be stored.
     comments : `~sunpy.io.header.FileHeader`, optional
         The comments to be stored as a header.
-    compress : `int`, optional
-        To compress the data or not.
-        1 is to compress, 0 is uncompressed
+    compress : `int`, {0, 1} optional
+        To compress the data or not with ``1`` to compress and ``0`` is uncompressed.
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy/io/ana.py]: It probably should but I would have to check if works with the C code. This probably should be done when I update the code to work with windows.

[the remark this answers is not in the document]
Comment (dpshelio) [reply sunpy/io/ana.py]: The python interface should be boolean, then converted to whatever it needs in the back.

[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy/io/ana.py]: Should be nicer now.

File: sunpy/io/ana.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy/io/ana.py]: Any thoughts?

File: sunpy/io/ana.py [hunk 4]
```diff
@@ -108,11 +108,16 @@ def write(filename, data, comments=False, compress=1, debug=False):
...
-    >>> written = sunpy.io.ana.write(filename, data, comments=False, compress=1)  # doctest: +SKIP
+    >>> written = sunpy.io.ana.write(filename, data, comments=False, compress=True)  # doctest: +SKIP
     """
     if _pyana is None:
         raise ImportError("C extension for ANA is missing, please rebuild")
 
+    if compress:
+        compress = 1
+    else:
+        compress = 0
```
Comment (dpshelio) [inline sunpy/io/ana.py]: Or simply:

```python
compress *= 1
```

File: sunpy/io/ana.py [hunk 5]
```diff
@@ -96,10 +96,10 @@ def write(filename, data, comments=False, compress=1, debug=False):
         The data to be stored.
     comments : `~sunpy.io.header.FileHeader`, optional
         The comments to be stored as a header.
-    compress : `int`, {0, 1} optional
-        To compress the data or not with ``1`` to compress and ``0`` is uncompressed.
+    compress : `bool`, optional
+        Compress the data with `True` (the default).
```
Comment (dpshelio) [inline sunpy/io/ana.py]: :tada:

File: sunpy/io/ana.py (hunk 4 above)
Comment (Cadair) [reply sunpy/io/ana.py]: also `compress = int(compress)` would have worked here right?

File: .circleci/config.yml [hunk 6]
```diff
@@ -125,7 +125,7 @@ jobs:
       - run: *merge-check
       - run: *apt-install
       - run: *tox-install
-      - run: tox -e build_docs
+      - run: tox -e build_docs -- -j=auto
```
Comment (Cadair) [inline .circleci/config.yml]: parallel docs :scream_cat:

File: sunpy/io/ana.py [hunk 7]
```diff
@@ -1,29 +1,22 @@
...
------
-ANA is a script that allows people to access compressed ana files.
-It accesses a C library, based on Michiel van Noort's
-IDL DLM library 'f0' which contains a cleaned up version of the original
-anarw routines.
+.. warning::
 
-Created by Tim van Werkhoven (t.i.m.vanwerkhoven@gmail.com) on 2009-02-11.
-Copyright (c) 2009--2011 Tim van Werkhoven.
+    The reading and writing of ana file is not supported under Windows.
```
Comment (Cadair) [inline sunpy/io/ana.py]:
```suggestion
    The reading and writing of ana files is not supported under Windows.
```

File: .circleci/config.yml (hunk 6 above)
Comment (nabobalis) [reply .circleci/config.yml]: It didn't really do anything. Not sure if the setting is wrong or something else is up.

File: sunpy/io/ana.py [hunk 8]
```diff
@@ -115,10 +108,12 @@ def write(filename, data, comments=False, compress=1, debug=False):
...
     Examples
     --------
-    >>> written = sunpy.io.ana.write(filename, data, comments=Falsem, compress=1)   # doctest: +SKIP
+    >>> written = sunpy.io.ana.write(filename, data, comments=False, compress=True)  # doctest: +SKIP
     """
     if _pyana is None:
-        raise ImportError("C extension for ANA is missing, please rebuild")# pragma: no cover
+        raise ImportError("C extension for ANA is missing, please rebuild")
+
+    compress *= 1
```
Comment (Cadair) [inline sunpy/io/ana.py]: This feels like a weird hack, I would say it either needs a comment, or a better way would be to just do `int(compress)` in the function call below.

File: sunpy/io/file_tools.py [hunk 9]
```diff
@@ -1,6 +1,8 @@
-import re
+"""
+This module provides a generic file reader.
+"""
 import os
-import collections
+import re
```
Comment (Cadair) [inline sunpy/io/file_tools.py]: What we using `re` for here?

File: sunpy/io/tests/test_ana.py [hunk 10]
```diff
@@ -64,18 +60,18 @@ def test_f32u():
...
-    assert np.sum(img_f32u_rec[0][0]- img_f32) == 0
+    assert np.sum(img_f32u_rec[0][0] - img_f32) == 0
 
 
-@skip_ana
-def test_f32c():
-    # Test if float 32 compressed functions
-    #TODO: Bug with same code. Needs to be tracked down.
-
+# TODO: Bug with same code. Needs to be tracked down.
```
Comment (Cadair) [inline sunpy/io/tests/test_ana.py]: Is this why the coverage has dropped? Why is this breaking?

Comment (nabobalis) [reply sunpy/io/tests/test_ana.py]: Yeah, this is probably why the coverage is down. I'm not sure why, clearly something I didn't fix when we merged in the code, way back when.

**Comment by nabobalis** [comment]
I'll fix this up and rebase the PR this evening.

File: sunpy/io/file_tools.py (hunk 9 above)
Comment (nabobalis) [reply sunpy/io/file_tools.py]: In _detect_filetype

**Comment by nabobalis** [comment]
Should be good to go now.

**Pull request merged**